### PR TITLE
test_e2e: Add container engine validation

### PR DESCRIPTION
**Pull request by dougsland**
Developers might not use only docker but different
containers engine like podman. This patch adds support to it.

Signed-off-by: Douglas Schilling Landgraf <dlandgra@redhat.com>

### CODE REVIEW

File: hack/test_e2e.sh [hunk 1]
```diff
@@ -40,6 +41,44 @@ if_error_exit() {
...
+    #   None                                                                  #
+    ###########################################################################
+
+    echo "* Detecting container engine..."
+    # If docker is not available, let's check if podman exists
+    ${CONTAINER_ENGINE} &> /dev/null
+    if [ "$?" != "0" ]; then
+        CONTAINER_ENGINE="podman"
+        ${CONTAINER_ENGINE} --help &> /dev/null
+        if_error_exit "Unable to detect container engine, try docker or podman"
```
Comment (anusha94) [inline hack/test_e2e.sh:77]: How about phrasing the message as 
`The e2e tests currently support docker and podman as the container engine. Please install either of them`?

**Review by anusha94 — COMMENTED** [review]
lgtm.

File: hack/test_e2e.sh (hunk 1 above)
Comment (dougsland) [reply hack/test_e2e.sh:77]: make sense

Comment (dougsland) [reply hack/test_e2e.sh:77]: sure, make sense. thanks for the feedback.

**Review by nehaLohia27 — APPROVED** [review]

**Pull request merged**